pkgs/tools/networking/ngrok-2/default.nix

```diff
@@ -1,31 +1,38 @@
 { stdenv, fetchurl, unzip }:
 
+with stdenv.lib;
+
 stdenv.mkDerivation rec {
   name = "ngrok-${version}";
   version = "2.2.8";
 
-  src = if stdenv.system == "i686-linux" then fetchurl {
+  src = if stdenv.isLinux && stdenv.isi686 then fetchurl {
     url = "https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-i386.tgz";
     sha256 = "0s5ymlaxrvm13q3mlvfirh74sx60qh56c5sgdma2r7q5qlsq41xg";
-  } else if stdenv.system == "x86_64-linux" then fetchurl {
+  } else if stdenv.isLinux && stdenv.isx86_64 then fetchurl {
     url = "https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.tgz";
     sha256 = "1mn9iwgy6xzrjihikwc2k2j59igqmph0cwx17qp0ziap9lp5xxad";
+  } else if stdenv.isDarwin then fetchurl {
+    url = "https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-darwin-386.zip";
+    sha256 = "0yfd250b55wcpgqd00rqfaa7a82f35fmybb31q5xwdbgc2i47pbh";
   } else throw "platform ${stdenv.system} not supported!";
```

Comment:
I personally find attribute lookup to be a bit nicer for this

Reply:
you mean `stdenv.system == "x86_64-darwin"`?

Reply:
Like this ee6d0a7025df93ee93142de0c25264c586b125c7 instead of the if/else statements, but it's mostly personal preference you don't have to change it.

Reply:
ok good to known that